app/api/test/module/Api/src/Entity/User/UserEntityTest.php

```diff
@@ -1405,8 +1405,40 @@ public function dpCanResetPassword(): array
     public function dpOperatorAdminRoles(): array
     {
         return [
-            [RoleEntity::ROLE_OPERATOR_ADMIN],
-            [RoleEntity::ROLE_OPERATOR_TC],
+            [RoleEntity::ROLE_OPERATOR_ADMIN]
         ];
     }
+
+    public function testIsTransportConsultant()
+    {
+        $roleTC = m::mock(RoleEntity::class)->makePartial();
+        $roleTC->setRole(RoleEntity::ROLE_OPERATOR_TC);
+
+        $roleOther = m::mock(RoleEntity::class)->makePartial();
+        $roleOther->setRole('some_other_role');
+
+        $userTC = new UserEntity('pid', UserEntity::USER_TYPE_OPERATOR);
+        $userTC->setRoles(new ArrayCollection([$roleTC]));
+        $this->assertTrue($userTC->isTransportConsultant(), 'User should be identified as a transport consultant');
+
+        $userNonTC = new UserEntity('pid', UserEntity::USER_TYPE_OPERATOR);
```

Comment:
test is checking that user is TC (cannot get over fact we use TC to mean other things too but that is me) when assigned but also assertion to say they are not - not a blocker but better as 2 tests ? testIsTransportConsultant
testTransportConsultantRoleNotDefault  I assume that is why asseertion we are ensuring role is not default ?

Reply:
After some discussion with Ian, looks like a new UserType will not work well with some existing validation rules elsewhere in the app, so shelving these changes. Thanks for the feedback though!